### PR TITLE
metricbeat: suppress error when RAID metrics are enabled on non-RAID system

### PR DESCRIPTION
**Pull request by mauri870**
## Proposed commit message

When the Linux Metrics integration (beta) is installed with the raid metrics option enabled it causes an error if the host does not have a RAID configuration. This error causes the Agent to go into a degraded state.

This happens because we report not having `/sys/block/md*` devices  as an error, this only means that no RAID configuration is in place.

Instead, do not report this case as an error.

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Closes https://github.com/elastic/elastic-agent/issues/6155

### CODE REVIEW

File: metricbeat/module/system/raid/blockinfo/getdev.go [hunk 1]
```diff
@@ -19,14 +19,13 @@ package blockinfo
 
 import (
 	"fmt"
-	"io/ioutil"
 	"os"
 	"path/filepath"
 )
 
 // ListAll lists all the multi-disk devices in a RAID array
 func ListAll(path string) ([]MDDevice, error) {
```
Comment (VihasMakwana) [inline metricbeat/module/system/raid/blockinfo/getdev.go:27]:
```suggestion
func ListAll(logger *logp.Logger, path string) ([]MDDevice, error) {
```

File: metricbeat/module/system/raid/blockinfo/getdev.go [hunk 2]
```diff
@@ -44,7 +43,7 @@ func ListAll(path string) ([]MDDevice, error) {
 	}
 
 	if len(mds) == 0 {
-		return nil, fmt.Errorf("no matches from path %s", path)
+		return nil, nil
```
Comment (VihasMakwana) [inline metricbeat/module/system/raid/blockinfo/getdev.go:46]:
```suggestion
		logger.Warn("No match found. You've probably enabled RAID module on a non-RAID system")
		return nil, nil
```

**Review by VihasMakwana — COMMENTED** [review]
I've added a few suggestions.
Also, need to all instances of `ListAll` function calls if you add logs.

**Comment by VihasMakwana** [comment]
Or there is alternative.

Return the array, but as a `PartialMetricsError` so the agent will stay healthy but the message will be reflected on console if they do `elastic-agent status`
https://github.com/elastic/beats/blob/4942592a8a272b5355c0be531e5f052b2c7a1d56/metricbeat/mb/module/wrapper.go#L313-L324

**Comment by mauri870** [comment]
> Or there is an alternative.
> 
> Return the error, but as a `PartialMetricsError` so the agent will stay healthy but the message will be reflected on console if they do `elastic-agent status`

Thanks! I like this idea better. It aligns well with the problem we’re addressing and clearly communicates to the user that they need to take action. Displaying this in the agent status will prompt the user to disable the RAID option, which makes sense since it has no effect. I'll address this here.

File: metricbeat/module/system/raid/blockinfo/getdev.go (hunk 1 above)
Comment (mauri870) [reply metricbeat/module/system/raid/blockinfo/getdev.go:27]: Resolving since I'll go with the partial metrics error solution.

**Review by VihasMakwana — APPROVED** [review]

**Pull request merged**